Modules/POSIX/Tests/POSIXTests/RegexTests.swift

```diff
@@ -0,0 +1,204 @@
+        XCTAssertThrowsError(try Regex(pattern: "*"))
+    }
+
+    func testMatches() throws {
+        let regex = try Regex(pattern: "hello")
+        let matches = regex.matches(in: "hello")
+        XCTAssert(matches)
+    }
+
+    func testNotMatches() throws {
+        let regex = try Regex(pattern: "hello")
+        let matches = regex.matches(in: "bye")
+        XCTAssert(!matches)
+    }
+
+    func testGroup() throws {
+        let regex = try Regex(pattern: "(hello)")
+        let groups = regex.groups(in: "hello")
+        XCTAssert(groups == ["hello"])
+    }
+
+    func testGroups() throws {
+        let regex = try Regex(pattern: "(hello) (world)")
+        let groups = regex.groups(in: "hello world")
+        XCTAssert(groups == ["hello", "world"])
+    }
+
+    func testNoGroups() throws {
+        let regex = try Regex(pattern: "(hello)")
+        let groups = regex.groups(in: "bye")
+        XCTAssert(groups == [])
+    }
+
+    /// Replace one occurence by a shorter template string
+    func testReplaceOneOccurenceWithShorterTemplate() throws {
+        let regex = try Regex(pattern: "hello")
+        let string = regex.replace(in: "hello world", with: "bye")
+        XCTAssert(string == "bye world")
+    }
+
+    /// Replace one occurence by a longer template string
+    func testReplaceOneOccurenceWithLongerTemplate() throws {
+        let regex = try Regex(pattern: "o")
+        let string = regex.replace(in: "hello", with: "ooo!")
+        XCTAssertEqual(string, "hellooo!")
+    }
+
+    /// Replace multiple occurences each by a shorter template string
+    func testReplaceManyOccurencesWithShorterTemplate() throws {
+        let regex = try Regex(pattern: "mm")
+        let string = regex.replace(in: "mm-mm", with: "M")
+        XCTAssertEqual(string, "M-M")
+    }
+
+    /// Replace multiple occurences each by a longer template string
+    func testReplaceManyOccurencesWithLongerTemplate() throws {
+        let regex = try Regex(pattern: "l")
+        let string = regex.replace(in: "lol", with: "LL")
+        XCTAssertEqual(string, "LLoLL")
+    }
+
+    /// Replace each digit
+    func testReplaceDigits() throws {
+        let regex = try Regex(pattern: "[0-9]")
+        let string = regex.replace(in: "1234", with: ".")
+        XCTAssertEqual(string, "....")
+    }
+
+    /// Replace number
+    func testReplaceNumber() throws {
+        let regex = try Regex(pattern: "[0-9]+")
+        let string = regex.replace(in: "1234", with: ".")
+        XCTAssertEqual(string, ".")
+    }
+
+    func testNoReplace() throws {
+        let regex = try Regex(pattern: "bye")
+        let string = regex.replace(in: "hello world", with: "bye")
+        XCTAssert(string == "hello world")
+    }
+    
+    func testReplaceUTF8() throws {
+        let r0 = try Regex(pattern: "coffee")
+        let actual0 = r0.replace(in: "Paulo loves coffee", with: "☕️")
+        XCTAssertEqual(actual0, "Paulo loves ☕️")
```

Comment:
I could easily use `String(bytes:encoding:)` but not `String(cString:)` because it needs a `CChar` and I'd rather avoid an `unsafeBitCast`. Do you agree?